lms/djangoapps/class_dashboard/dashboard_data.py

```diff
@@ -180,11 +187,29 @@ def get_d3_problem_grade_distrib(course_id):
                         # Construct label to display for this problem
                         label = "P{0}.{1}.{2}".format(c_subsection, c_unit, c_problem)
 
-                        # Only problems in prob_grade_distrib have had a student submission.
-                        if child.location in prob_grade_distrib:
+                        grade_distribution = []
+
+                        try:
+                            problem = client.modules(course_id, child.location)
```

Comment:
Yes, `try/catch` blocks should be defined as narrowly as possible: catch the smallest set of exceptions, specify caught exception types as specifically as possible, and not wrap code that can't throw the specified exception.